src/main/java/org/wildfly/security/sasl/util/SSLQueryCallbackHandler.java

```diff
@@ -56,7 +56,7 @@ public SSLQueryCallbackHandler(final CallbackHandler delegate, final Supplier<SS
 
     public void handle(final Callback[] callbacks) throws IOException, UnsupportedCallbackException {
         SSLSession sslSession;
-        if (! once.compareAndSet(false, true) || (sslSession = this.sslSession.get()) == null) {
+        if (! once.compareAndSet(true, false) || (sslSession = this.sslSession.get()) == null) {
```

Comment:
Yeah "once" is initialized to true, and when we successfully change it from true to false, we are first.  Though I think maybe the ! is wrong...